packages/wds-event-tool/src/main/kotlin/org/webdevandsausages/events/controller/PatchEvent.kt

```diff
@@ -0,0 +1,53 @@
+package org.webdevandsausages.events.controller
+
+import arrow.core.Either
+import arrow.core.Some
+import org.http4k.contract.ContractRoute
+import org.http4k.contract.div
+import org.http4k.contract.meta
+import org.http4k.core.*
+import org.webdevandsausages.events.Router
+import org.webdevandsausages.events.dto.EventUpdateInDto
+import org.webdevandsausages.events.dto.whenValid
+import org.webdevandsausages.events.error.toResponse
+import org.webdevandsausages.events.service.GetEventByIdService
+import org.webdevandsausages.events.service.UpdateEventService
+import org.webdevandsausages.events.utils.WDSJackson.auto
+
+object PatchEvent {
+    private val eventUpdateRequestLens = Body.auto<EventUpdateInDto>().toLens()
```

Comment:
Since http4k already does some basic validation here, I think we could also put our validation in the lens, like this `Body.json().map({ json -> EventUpdateInDto.from(json) }).toLens()` This is actually not that different from what you did, just happens early and kind of integrates more with how http4k handles errors in filters.  I suppose nother possibility is to send the data as form data in which case htt4k has really nice built in validator lenses (but I kind of dumb to convert it for just this reason I guess)l

Reply:
then the validation would naturally happen inside the `from` on the DTO. If the then throws validation errors, I think the lens filter should catch and put it in a response. I think you could reuse the validators you already made. It is an interesting library.

Looking at http4k, they return validation errors something like this:
```{  
   "message":"Missing/invalid parameters",
   "params":[  
      {  
         "name":"name1",
         "type":"query",
         "datatype":"string",
         "required":true,
         "reason":"Missing"
      },
      {  
         "name":"name2",
         "type":"header",
         "datatype":"number",
         "required":false,
         "reason":"Invalid"
      }
   ]
}
```